### ISSUE TITLE
The weird color and bad appearance results

### ISSUE THREAD

**Issue by A-pril** [Gorilla-Lab-SCUT/Fantasia3D#49]
Hi~ It's great work. But I wanna use this code to generate texture only. And the results are all very bad just like the below.
![image](https://github.com/user-attachments/assets/431df738-1a6e-48a5-822e-719ceec7a105)

The config is here:
  {
    "mode": "appearance_modeling",
    "base_mesh": "xxx/nascar.obj",
    "out_dir": "nascar-0",
    "text" : "A next gen nascar in red",
    "seed" : 40288,
    "random_textures": true,
    "iter": 2000,
    "coarse_iter": 0, 
    "save_interval": 100,
    "texture_res": [ 2048, 2048 ],
    "train_res": [512, 512],
    "batch": 3,
    "kd_min" : [0.03, 0.03, 0.03],
    "kd_max" : [0.97, 0.97, 0.97],
    "ks_min" : [0, 0.08, 0],
    "ks_max" : [0, 0.7, 0],
    "display": [{"latlong" : true}, {"bsdf" : "kd"}, {"bsdf" : "ks"}, {"bsdf" : "normal"}],
    "envmap": "data/irrmaps/mud_road_puresky_4k.hdr",
    "env_scale" : 2.0,
    "train_background": "black",
    "validate" : true,
    "add_directional_text": true,
    "camera_random_jitter": 0.2,
    "fovy_range": [25.71, 45], 
    "elevation_range": [-10, 45],
    "guidance_weight": 50,
    "sds_weight_strategy": 2,
    "early_time_step_range": [0.02, 0.98],
    "late_time_step_range": [0.02, 0.98]
}

The conda list is here:
_libgcc_mutex             0.1                        main    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
_openmp_mutex             5.1                       1_gnu    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
accelerate                1.1.0                    pypi_0    pypi
addict                    2.4.0                    pypi_0    pypi
aiofiles                  22.1.0                   pypi_0    pypi
aiosqlite                 0.20.0                   pypi_0    pypi
anyio                     4.6.2.post1              pypi_0    pypi
argon2-cffi               23.1.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
arrow                     1.3.0                    pypi_0    pypi
asttokens                 2.4.1                    pypi_0    pypi
attrs                     24.2.0                   pypi_0    pypi
babel                     2.16.0                   pypi_0    pypi
beautifulsoup4            4.12.3                   pypi_0    pypi
blas                      1.0                         mkl    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
bleach                    6.2.0                    pypi_0    pypi
ca-certificates           2024.9.24            h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
certifi                   2024.8.30                pypi_0    pypi
cffi                      1.17.1                   pypi_0    pypi
charset-normalizer        3.4.0                    pypi_0    pypi
comm                      0.2.2                    pypi_0    pypi
contourpy                 1.3.0                    pypi_0    pypi
cuda-cudart               12.4.127             h99ab3db_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-cudart_linux-64      12.4.127             hd681fbe_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-cupti                12.4.127             h6a678d5_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-libraries            12.4.1               h06a4308_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-nvrtc                12.4.127             h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-nvtx                 12.4.127             h6a678d5_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-opencl               12.4.127             h6a678d5_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-runtime              12.4.1               hb982923_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cuda-version              12.4                 hbda6634_3    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cycler                    0.12.1                   pypi_0    pypi
debugpy                   1.8.7                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
deprecation               2.1.0                    pypi_0    pypi
diffusers                 0.31.0                   pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
exceptiongroup            1.2.2                    pypi_0    pypi
executing                 2.1.0                    pypi_0    pypi
fastjsonschema            2.20.0                   pypi_0    pypi
filelock                  3.13.1           py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
fonttools                 4.54.1                   pypi_0    pypi
fqdn                      1.5.1                    pypi_0    pypi
fsspec                    2024.6.1         py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
gmp                       6.2.1                h295c915_3    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
gmpy2                     2.1.2            py39heeb90bb_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
huggingface-hub           0.26.2                   pypi_0    pypi
idna                      3.10                     pypi_0    pypi
imageio                   2.35.1                   pypi_0    pypi
imageio-ffmpeg            0.5.1                    pypi_0    pypi
importlib-metadata        8.5.0                    pypi_0    pypi
importlib-resources       6.4.5                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46306    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
ipykernel                 6.29.5                   pypi_0    pypi
ipython                   8.18.1                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                8.1.5                    pypi_0    pypi
isoduration               20.11.0                  pypi_0    pypi
jedi                      0.19.1                   pypi_0    pypi
jinja2                    3.1.4            py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
joblib                    1.4.2                    pypi_0    pypi
json5                     0.9.25                   pypi_0    pypi
jsonpointer               3.0.0                    pypi_0    pypi
jsonschema                4.23.0                   pypi_0    pypi
jsonschema-specifications 2024.10.1                pypi_0    pypi
jupyter-client            7.4.9                    pypi_0    pypi
jupyter-core              5.7.2                    pypi_0    pypi
jupyter-events            0.10.0                   pypi_0    pypi
jupyter-packaging         0.12.3                   pypi_0    pypi
jupyter-server            2.14.2                   pypi_0    pypi
jupyter-server-fileid     0.9.3                    pypi_0    pypi
jupyter-server-terminals  0.5.3                    pypi_0    pypi
jupyter-server-ydoc       0.8.0                    pypi_0    pypi
jupyter-ydoc              0.2.5                    pypi_0    pypi
jupyterlab                3.6.8                    pypi_0    pypi
jupyterlab-pygments       0.3.0                    pypi_0    pypi
jupyterlab-server         2.27.3                   pypi_0    pypi
jupyterlab-widgets        3.0.13                   pypi_0    pypi
kiwisolver                1.4.7                    pypi_0    pypi
ld_impl_linux-64          2.40                 h12ee557_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libabseil                 20240116.2      cxx17_h6a678d5_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libcublas                 12.4.5.8             h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libcufft                  11.2.1.3             h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libcufile                 1.9.1.3              h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libcurand                 10.3.5.147           h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libcusolver               11.6.1.9             h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libcusparse               12.3.1.170           h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libedit                   3.1.20230828         h5eee18b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libffi                    3.4.4                h6a678d5_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libgcc-ng                 11.2.0               h1234567_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libgomp                   11.2.0               h1234567_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libnpp                    12.2.5.30            h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libnvfatbin               12.4.127             h7934f7d_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libnvjitlink              12.4.127             h99ab3db_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libnvjpeg                 12.3.1.117           h6a678d5_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libprotobuf               4.25.3               he621ea3_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libstdcxx-ng              11.2.0               h1234567_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
markupsafe                2.1.3            py39h5eee18b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
matplotlib                3.9.2                    pypi_0    pypi
matplotlib-inline         0.1.7                    pypi_0    pypi
mistune                   3.0.2                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46344    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mkl-service               2.4.0            py39h5eee18b_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mkl_fft                   1.3.10           py39h5eee18b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mkl_random                1.2.7            py39h1128e8f_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mpc                       1.1.0                h10f8cd9_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mpfr                      4.0.2                hb69a4c5_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mpmath                    1.3.0            py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
nbclassic                 1.1.0                    pypi_0    pypi
nbclient                  0.10.0                   pypi_0    pypi
nbconvert                 7.16.4                   pypi_0    pypi
nbformat                  5.10.4                   pypi_0    pypi
ncurses                   6.4                  h6a678d5_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
nest-asyncio              1.6.0                    pypi_0    pypi
networkx                  3.2.1            py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
ninja                     1.11.1.1                 pypi_0    pypi
notebook                  6.5.7                    pypi_0    pypi
notebook-shim             0.2.4                    pypi_0    pypi
numpy                     1.26.4           py39h5f9d8c6_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
numpy-base                1.26.4           py39hb5e798b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
nvdiffrast                0.3.0                    pypi_0    pypi
ocl-icd                   2.3.2                h5eee18b_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
open3d                    0.14.1                   pypi_0    pypi
opencv-python             4.10.0.84                pypi_0    pypi
openssl                   1.1.1w               h7f8727e_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
overrides                 7.7.0                    pypi_0    pypi
packaging                 24.1                     pypi_0    pypi
pandas                    2.2.3                    pypi_0    pypi
pandocfilters             1.5.1                    pypi_0    pypi
parso                     0.8.4                    pypi_0    pypi
pexpect                   4.9.0                    pypi_0    pypi
pillow                    11.0.0                   pypi_0    pypi
pip                       24.2             py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
platformdirs              4.3.6                    pypi_0    pypi
prometheus-client         0.21.0                   pypi_0    pypi
prompt-toolkit            3.0.48                   pypi_0    pypi
psutil                    6.1.0                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.3                    pypi_0    pypi
pycparser                 2.22                     pypi_0    pypi
pygments                  2.18.0                   pypi_0    pypi
pyparsing                 3.2.0                    pypi_0    pypi
python                    3.9.15               h7a1cb2a_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
python-dateutil           2.9.0.post0              pypi_0    pypi
python-json-logger        2.0.7                    pypi_0    pypi
pytorch-cuda              12.4                 hc786d27_7    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
pytz                      2024.2                   pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
pyzmq                     26.2.0                   pypi_0    pypi
readline                  8.2                  h5eee18b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
referencing               0.35.1                   pypi_0    pypi
regex                     2024.9.11                pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
rfc3339-validator         0.1.4                    pypi_0    pypi
rfc3986-validator         0.1.1                    pypi_0    pypi
rpds-py                   0.20.1                   pypi_0    pypi
safetensors               0.4.5                    pypi_0    pypi
scikit-learn              1.5.2                    pypi_0    pypi
scipy                     1.13.1                   pypi_0    pypi
send2trash                1.8.3                    pypi_0    pypi
setuptools                75.1.0           py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
six                       1.16.0                   pypi_0    pypi
sniffio                   1.3.1                    pypi_0    pypi
soupsieve                 2.6                      pypi_0    pypi
sqlite                    3.45.3               h5eee18b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
stack-data                0.6.3                    pypi_0    pypi
sympy                     1.13.2           py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
tbb                       2021.8.0             hdb19cb5_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
terminado                 0.18.1                   pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tinycss2                  1.4.0                    pypi_0    pypi
tinycudann                1.7                      pypi_0    pypi
tk                        8.6.14               h39e8969_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
tokenizers                0.20.2                   pypi_0    pypi
tomli                     2.0.2                    pypi_0    pypi
tomlkit                   0.13.2                   pypi_0    pypi
torch                     1.12.1+cu113             pypi_0    pypi
torchvision               0.13.1+cu113             pypi_0    pypi
tornado                   6.4.1                    pypi_0    pypi
tqdm                      4.66.6                   pypi_0    pypi
traitlets                 5.14.3                   pypi_0    pypi
transformers              4.46.1                   pypi_0    pypi
types-python-dateutil     2.9.0.20241003           pypi_0    pypi
typing_extensions         4.11.0           py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
tzdata                    2024.2                   pypi_0    pypi
uri-template              1.3.0                    pypi_0    pypi
urllib3                   2.2.3                    pypi_0    pypi
wcwidth                   0.2.13                   pypi_0    pypi
webcolors                 24.8.0                   pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.8.0                    pypi_0    pypi
wheel                     0.44.0           py39h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
widgetsnbextension        4.0.13                   pypi_0    pypi
xatlas                    0.0.9                    pypi_0    pypi
xz                        5.4.6                h5eee18b_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
y-py                      0.6.2                    pypi_0    pypi
ypy-websocket             0.8.4                    pypi_0    pypi
zipp                      3.20.2                   pypi_0    pypi
zlib                      1.2.13               h5eee18b_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main

I think it's abnormal. But how to fix it?

**Comment by zhulongcc** [comment]
meet a samilar problem. try a detailed text prompt may be useful for ur situation. im not sure whether it works or not but u can have an easy try:)

**Comment by A-pril** [comment]
> meet a samilar problem. try a detailed text prompt may be useful for ur situation. im not sure whether it works or not but u can have an easy try:)

Thanks for your reply. Here my prompt is "A next gen nascar in red", could u offer a detailed prompt for me? The prompt for any models is alright. I just wanna know the extent of detailed.

**Comment by zhulongcc** [comment]
i havent ur .obj file, so i provide a text and wish u try it. 
i generate a img using ur text "A next gen nascar in red" by sd and get a img:
![car](https://github.com/user-attachments/assets/ec711f10-e6ad-48f7-9394-d7d3f78137e1)
then get a description text by chatgpt:
Red race car featuring "22," sponsor logos, black wheels, and rear spoiler.
wish it works~

**Comment by A-pril** [comment]
Thanks for your patient reply. I will try this prompt soon~

**Comment by zhulongcc** [comment]
u r welcome~
